src/main/java/hudson/plugins/git/browser/GitLab.java

```diff
@@ -134,6 +135,7 @@ public URL getFileLink(Path path) throws IOException {
     }
 
     @Extension
+    @Symbol("gitLabBrowser")
```

Comment:
this is a change that would impact in the yaml configuration for CasC. So far, the exportation / configuration would have expected "gitLab": https://github.com/jenkinsci/git-plugin/blob/18d14466c63f62fdf070258eb35a97f6cb22513b/src/test/resources/jenkins/plugins/git/browsers-casc.yaml#L139
But with this change, the key word would be "gitLabBrowser". The former yaml configuration files have to be changed.

Reply:
CasC is able to recognize several values for the same descriptor, but it should be verified both of them work now.

Reply:
good to know! I thought CasC would only take into account one of them. That explains why browsers-casc.yaml didn't have to be updated

Reply:
This specific example is more complicated than you might expect (and thus even more interesting to assure that retain compatibility).  

The GitLab plugin provides a repository browser with class name GitLabBrowser.  It takes no arguments and appears in the pick list as "GitLab".  

The Git plugin provides a repository browser with a class name GitLab.  It takes one mandatory argument and an optional argument.

I assumed (possibly incorrectly) that the JCasC symbol "gitLabBrowser" was derived from the repository browser provided by the GitLab plugin, not from the repository browser provided by the Git plugin.

Reply:
> CasC is able to recognize several values for the same descriptor, but it should be verified both of them work now.

That gives me hope.  We might be able to define a symbol for compatibility with existing JCasC YAML files and also define a shorter or clearer symbol as the preferred symbol.

What types of things would we need to test to confirm that works as expected?

* JCasC config file for git plugin 4.2.2 (current release) loads correctly into new version
* JCasC config file for git plugin 4.3 (next release) exports correctly from JCasC page
* JCasC config file for git plugin 4.3 (next release) loads correctly
* Pipeline definitions from git plugin 4.2.2 (current release) load correctly in new version
* Pipeline syntax editor for git plugin 4.3 (next release)

Reply:
> What types of things would we need to test to confirm that works as expected?

Regarding the JCasC testing, that's what have to be done. Regarding the pipeline, I'm not sure

> I assumed (possibly incorrectly) that the JCasC symbol "gitLabBrowser" was derived from the repository browser provided by the GitLab plugin, not from the repository browser provided by the Git plugin.

The `BrowsersJCasCCompatibilityTest` (via the `browsers-casc.yaml`) is testing the former `gitlab` key word and it's working. In a production environment where we have installed the gitlab-plugin and the git-plugin, if we can have both repository browsers, we have to check which key word is exported for each case. If they are not the same (gitLabBrowser vs gitlab), there should be no problem, since the configurators will be able to distinguish between them. But if they are the same with this new Symbol, then I foresee we'd get an error. I guess this check must be done with the current release and with this PR merged

Reply:
CasC as of a few versions now considers all symbols on import and on export will show the first one in the array